Zend/zend_inheritance.c

```diff
@@ -1139,6 +1188,12 @@ static void do_inherit_class_constant(zend_string *name, zend_class_constant *pa
 	} else if (!(Z_ACCESS_FLAGS(parent_const->value) & ZEND_ACC_PRIVATE)) {
 		if (Z_TYPE(parent_const->value) == IS_CONSTANT_AST) {
 			ce->ce_flags &= ~ZEND_ACC_CONSTANTS_UPDATED;
+			ce->ce_flags |= ZEND_ACC_HAS_AST_CONSTANTS;
+			if (ce->parent->ce_flags & ZEND_ACC_IMMUTABLE) {
+				c = zend_arena_alloc(&CG(arena), sizeof(zend_class_constant));
+				memcpy(c, parent_const, sizeof(zend_class_constant));
+				parent_const = c;
+			}
```

Comment:
I don't really get why this copy is needed.

Reply:
We can't update constant value in shared memory, so we copy it into process memory.
If you comment this some tests are going to fail. For example tests/classes/constants_basic_006.phpt